lib/Serialization/Serialization.cpp

```diff
@@ -1297,15 +1299,27 @@ void Serializer::writeInputBlock() {
     off_t importedHeaderSize = 0;
     time_t importedHeaderModTime = 0;
     std::string contents;
-    if (!Options.ImportedHeader.empty()) {
+    auto importedHeaderPath = Options.ImportedHeader;
+    // We do not want to serialize the explicitly-specified .pch path if one was
```

Comment:
Do we have test coverage for this? That is what happens when `-disable-bridging-pch` is used?

Reply:
`header_deps_of_binary.swift` is testing this very thing. A binary module is being built with a direct `.pch` file and what is serialized in it is a `.h` of the source. 

`-disable-bridging-pch` is a driver flag to always feed the `.h` directly to the TU compilation, instead of ever creating a PCH.